test/harness/asyncHelpers-asyncTest-return-not-thenable.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: |
+    The 'asyncTest' helper rejects test functions that do not return a thenable.
+includes: [asyncHelpers.js, compareArray.js]
+---*/
+
+const doneValues = [];
+function $DONE(error) {
+  doneValues.push(error instanceof TypeError);
```

Comment:
The RFC doesn't say what kind of error this is, but I'd guess it should be a `Test262Error` in parallel with `assert.throwsAsync()`.

Reply:
I'm not sure I get where this TypeError is thrown, it seems to me that it should be a Test262Error according to the following code:
```js
innerThenable = func();
if (
  innerThenable === null ||
  typeof innerThenable !== "object" ||
  typeof innerThenable.then !== "function"
) {
  message += /* etc */;
  throw new Test262Error(message);
}
```
it seems to me that e.g. `asyncTest(() => 42)` assigns 42 to `innerThenable`, so the condition should trigger here. But the test passes, so I'm not sure what's going on.

Reply:
> The RFC doesn't say what kind of error this is, but I'd guess it should be a `Test262Error` in parallel with `assert.throwsAsync()`.

Oops, I forgot to respond to this rather than just mark the issue resolved, sorry! That is very much my mistake. Thank you for bringing it up again. It comes from [this line](https://github.com/tc39/test262/blob/main/rfcs/async-helpers.md?plain=1#L103) - this occurs when calling testFunc synchronously throws. In this case, because testFunc is not a function, trying to call it throws a TypeError. The test passes as a result, and if we want to change it to throw a Test262Error I'd support that - but we should change the RFC text to match.

Reply:
I see where I was confused now; I was looking at the code for `assert.throwsAsync()`, but this is `asyncTest()`.